File: check-view-columns.js
import { createClient } from '@supabase/supabase-js';
import * as dotenv from 'dotenv';
dotenv.config();

const supabaseUrl = process.env.SUPABASE_URL;
const supabaseKey = process.env.SUPABASE_SERVICE_ROLE_KEY;

const supabase = createClient(supabaseUrl, supabaseKey);

async function checkViewColumns() {
  console.log('Checking columns in athletes_with_waiver_status view...');
  
  try {
    // Get a sample row from the view
    const { data, error } = await supabase
      .from('athletes_with_waiver_status')
      .select('*')
      .limit(1);
    
    if (error) {
      console.error('Error fetching data:', error);
      return;
    }
    
    if (!data || data.length === 0) {
      console.log('No data found in the view');
      return;
    }
    
    // Get the column names from the first row
    const columns = Object.keys(data[0]);
    console.log('Columns in athletes_with_waiver_status view:');
    columns.forEach(col => console.log(`- ${col}`));
    
    // Check for waiver-related columns
    const waiverColumns = columns.filter(col => col.includes('waiver'));
    console.log('\nWaiver-related columns:');
    waiverColumns.forEach(col => console.log(`- ${col}`));
    
    // Show sample data for specific columns we're interested in
    console.log('\nSample data for waiver columns:');
    waiverColumns.forEach(col => {
      console.log(`${col}: ${JSON.stringify(data[0][col])}`);
    });
    
    console.log('\nFull sample data:');
    console.log(JSON.stringify(data[0], null, 2));
  } catch (err) {
    console.error('Exception:', err);
  }
}

checkViewColumns();

File: Tests/python/check-view-columns.py
import os
from dotenv import load_dotenv
import psycopg2
import json
import urllib.parse

# Load environment variables
load_dotenv()

# Try different environment variables for database connection
db_url = os.getenv('DATABASE_URL')
if not db_url:
    # Try the direct URL
    db_url = os.getenv('DATABASE_DIRECT_URL')
    if db_url:
        print(f"Using DATABASE_DIRECT_URL: {db_url[:30]}...")

try:
    # Connect to the database
    conn = psycopg2.connect(db_url)
    cur = conn.cursor()

    # Query to get columns from the view
    cur.execute("""
        SELECT column_name
        FROM information_schema.columns
        WHERE table_name = 'athletes_with_waiver_status'
        ORDER BY ordinal_position;
    """)
    
    columns = [col[0] for col in cur.fetchall()]
    print("Available columns in athletes_with_waiver_status view:")
    for col in columns:
        print(f"- {col}")
    
    # Query to get sample data
    cur.execute("""
        SELECT *
        FROM athletes_with_waiver_status
        WHERE waiver_relationship_to_athlete IS NOT NULL
        LIMIT 1;
    """)
    
    row = cur.fetchone()
    if row:
        # Create a dictionary of column names and values
        sample_data = {}
        for i, col in enumerate(columns):
            sample_data[col] = row[i]
        
        print("\nSample data for relationship_to_athlete:")
        if 'waiver_relationship_to_athlete' in sample_data:
            print(f"waiver_relationship_to_athlete: \"{sample_data['waiver_relationship_to_athlete']}\"")
        else:
            print("waiver_relationship_to_athlete column not found in results")
            
        print("\nRelated waiver fields:")
        for key in sample_data:
            if key.startswith('waiver_'):
                print(f"{key}: \"{sample_data[key]}\"")
                
        print("\nFull sample data:")
        print(json.dumps(sample_data, indent=2, default=str))
    else:
        print("No sample data found with non-null relationship_to_athlete")
    
    cur.close()
    conn.close()

except Exception as e:
    print(f"Error: {e}")
